web/src/app/models/survey.model.ts

```diff
@@ -14,6 +14,14 @@
  * limitations under the License.
  */
 
+/**
+ * Represents the state of the survey, undefined should be considered as DONE.
+ */
+export enum State {
+  INITIAL_SETUP,
+  DONE,
```

Comment:
Can we use `null` instead of `DONE` for backwards (and forwards) compatibility as discussed?

Reply:
As stated in the comments, `undefined` should be considered as `DONE`. I like to have both `undefined` and `DONE` that both means `DONE`, we should use `DONE` explicitly and allow `undefined` in the meantime, also the enum `DONE` serves as an anchor for `undefined` to related to.

Reply:
The problem is that we don't know yet what future states will be, but it's fairly certain "DONE" will be confusing once we introduce new ones. In general it's more robust to have only one way to represent a particular state. Did you want this for use in switch statements? In those cases you could also use the default branch of switch to cover this case.

Reply:
No, this is not for template, we already agreed on extracting what's for UI and not store that in the backend.

Why is `DONE` confusing? To me, having to remember `undefined` means `DONE` while there is only one state in this enum called `INITIAL_SETUP ` is not straightforward.

Reply:
That's also true. But it may be misleading or unclear that the survey is in a "Done" state since it's specific to the survey creation flow. I wonder if it's not just simpler and clearer to only show the create flow when one of the required elements in survey are missing? If someone leaves when they're on the "review" page, they'd just get bounced to the "edit" flow, which isn't necessarily the wrong thing to do either. My thinking here is I'm trying to avoid needing the data model to reflect anything specific to the creation flow, but rather having the creation flow react to the data in the survey. Curious to hear what you think.

Reply:
Chatted over GVC, we decided to remove the `SurveyState` and judge whether the survey is still `INITIAL_SETUP` or not based on the content of the survey.